climate-monitoring-core/src/main/java/climatemonitoring/core/ViewState.java

```diff
@@ -1,6 +1,35 @@
+/*
+
+Alessandro della Frattina 753073 VA
+Cristian Capiferri 752918 VA
+Francesco Lops 753175 VA
+Dariia Sniezhko 753057 VA
+
+*/
+
 package climatemonitoring.core;
 
+/**
+ * Abstract class that is used to represent all views of the application
+ * @author francescolops
+ * @version 1.0-SNAPSHOT
+ */
 public abstract class ViewState {
 
+	/**
+	 * Code that runs in headless configuration only
+	 * @param args
+	 */
+	public void onHeadlessRender(String args){
+
+	}
+
+	/**
+	 * Code that runs in GUI configuration only
+	 */
+	public void onGUIRender(){
+		
+	}
```

Comment:
These methods should be abstract, so you should not provide any definition for them. You should also remove the `@param` tag if you decide to leave it blank. Example:
```
public abstract void onHeadlessRender(String args);
public abstract void onGUIRender();
```